app/controllers/merchant/items_controller.rb

```diff
@@ -6,13 +6,20 @@ def index
   end
 
   def new
+    merchant = current_user.merchant
+    @item = merchant.items.new(item_params)
   end
 
   def create
     merchant = current_user.merchant
-    merchant.items.create!(item_params)
-    flash[:mesage] = "Your new item has been saved and is now active and available for sale."
-    redirect_to '/merchant/items'
+    @item = merchant.items.new(item_params)
+    if @item.save
+      flash[:mesage] = "Your new item has been saved and is now active and available for sale."
+      redirect_to '/merchant/items'
+    else
+      flash.now[:errors] = @item.errors.full_messages.uniq.to_sentence
+      render :new
```

Comment:
We also had to use render :new here instead of re-direct for the pre-population of the form.

Reply:
Good to see the different uses for render/redirect